### PR TITLE
Inspector: Implemented a collapse/expand all

### LINKED ISSUE #9427: Option to expand inspector sections by default
**Godot version: Master 36548a3fa**

**Issue description:**
The new collapsible sections on the inspector are cool, but all start collapsed and if you need to find a specific property you need to remember to which class it belongs or expand all, every time (depending on the volatility of the user's memory).
This may not help new users either.

An option for having sections expanded by default on the inspector could be useful.

### PR DESCRIPTION
**Pull request by Krakean**
Implemented a collapse all / expand all feature request (fixes #9427)
This is how it works in action: https://i.imgur.com/RTmLaPh.gifv

Pretty small addition, hopefully will be added to 3.0 :)

P.S. Thought about to implement it as one-switch button, but it would require a much of unnecessary changes for 3.0-beta considering that we are in feature freeze state. Current solution is pretty simple and neat, in my opinion. And do not introduce any issues :)

### CODE REVIEW

File: editor/property_editor.h [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿/*************************************************************************/
+/*************************************************************************/
```
Comment (akien-mga) [inline editor/property_editor.h]: Your IDE added a BOM here, could you amend the commit to remove it?

Comment (Krakean) [reply editor/property_editor.h]: Can you please manually fix it? Can't see a BOM, nor in Visual Studio nor in Notepad++ nor in AkelPad. Shows me typical UTF-8 without any bom.

Comment (Krakean) [reply editor/property_editor.h]: And clang-format also doesn't fix anything.

Comment (akien-mga) [reply editor/property_editor.h]: Nevermind, it looks like you *removed* an existing BOM, so that's good.

**Comment by djrm** [comment]
im not sure about this one, two buttons dont seem right, i think it would be better to add a context menu with `expand all` and `collapse all` to tree, that way every tree will benefit from this. (the context menu could be activated by right clicking on section titles or something like that)

**Comment by Krakean** [comment]
@djrm 
Context menu doesn't seems right to me, it adds one more mouse click which is unneeded in this situation in my opinion. 
If you think that two buttons doesn't seems right anyway, well, one button is not solution too. 
Simple case: https://i.imgur.com/7gCjCfk.png -- what icon I should display in this case, when just one or two properties are expanded -- expand all or collapse all?

**Comment by djrm** [comment]
i never said one button that doesnt makes sense, what i was suggesting was a less intrusive more standarized way to access such functionality, it would be useful in other trees, like input map, also one more press is not bad for a secondary action like this (and its already saving you a ton of presses).

**Comment by Krakean** [comment]
@djrm 
> standarized way to access such functionality, it would be useful in other trees, like input map

Will think about adding such funcitonality to other places too :)
But first of all would be nice if this PR will be merged, as for start.

**Comment by Dillybob92** [comment]
I was thinking of global setting that we change in Editor Settings. Either collapse inspectors or not (Boolean), but I mean to get this before feature freeze would be ideal too :D Awesomeeee!!

**Comment by Krakean** [comment]
@Dillybob92 current my implementation of this feature allows easily to add an Editor Settings-level option, so this is something I keep in mind if  "collapse/expand all" feature will be actually used & welcomed by users in general.

**Comment by reduz** [comment]
I think this should be implemented as follows:

1. Move the buttons to the menu:

image

2. Add a project setting to make property editor open unfolded on default (if this does not exist already)

The rationale is that, if you will be opening this all the time, you may as well make it the default. If you will do it ocassionally, then you don't need a button visible all the time.

**Comment by Krakean** [comment]
@reduz
1. Thought about that, but decided to position them in order of importance from low to highest, as I understand importance of "show search" and "object properties" buttons. But anyway, will do as you say, not a problem.
2. Okay, will add an editor setting for it. Didn't made it initially due to feature freeze state, so thought PR could be not accepted if I add more than necessary. 

PR will be updated in two days, at the moment my PC in service center.

**Comment by groud** [comment]
From what I understand, feature freeze does not include usability improvements. So it's ok merging this :)

**Comment by reduz** [comment]
@Krakean you have until the weekend for beta, afterwards we won't merge any more PRs that are not fixes

**Comment by Krakean** [comment]
@reduz
Got it.

**Comment by Krakean** [comment]
Reimplemented PR as #13233

**Comment by akien-mga** [comment]
Superseded by #13233.

**Pull request closed without merging**